adb_graphics/datahandler/grib.py

```diff
@@ -549,17 +682,35 @@ def values(self, level=None, name=None, **kwargs) -> np.ndarray:
                 raise errors.NoGraphicsDefinitionForVariable(name, level)
             field = self.get_field(ncl_name or self.ncl_name(spec))
 
-        if len(field.shape) == 2:
-            vals = field[::]
+        lev = vertical_index
+        vertical_dim = []
 
-        elif len(field.shape) == 3:
-            if one_lev:
-                lev = vertical_index
-                if vertical_index is None:
+        vals = field
+        if one_lev:
+
+            # Check if it's a 3D variable (lv in any dimension field)
+            vertical_dim = [dim for dim in field.dims if 'lv' in dim]
+
+            if vertical_dim: # Field has a vertical dimension
+
+                dim_name = spec.get('vertical_level_name', vertical_dim[0])
+                dim_name = vertical_dim[0]
+
+                if vertical_index is None: # No index is provided in kwargs
                     lev = self.get_level(field, level, spec)
-                vals = field[lev, :, :]
-            else:
-                vals = field[:, :, :]
+
+                try:
+                    vals = vals.isel(**{dim_name: lev})
```

Comment:
This is a more pythonic way of handling grabbing a slice of data, especially when we don't know the dimension order or possible dimensions of the datasets.